test/lib/jdk/test/lib/Platform.java

```diff
@@ -260,28 +260,46 @@ public static boolean hasSA() {
         return true;
     }
 
-    /**
-     * Return true if the test JDK is hardened, otherwise false. Only valid on OSX.
-     */
-    public static boolean isHardenedOSX() throws IOException {
-        // We only care about hardened binaries for 10.14 and later (actually 10.14.5, but
-        // for simplicity we'll also include earlier 10.14 versions).
-        if (getOsVersionMajor() == 10 && getOsVersionMinor() < 14) {
-            return false; // assume not hardened
-        }
-
-        // Find the path to the java binary.
+    private static Process launchCodesignOnJavaBinary() throws IOException {
         String jdkPath = System.getProperty("java.home");
         Path javaPath = Paths.get(jdkPath + "/bin/java");
         String javaFileName = javaPath.toAbsolutePath().toString();
         if (Files.notExists(javaPath)) {
             throw new FileNotFoundException("Could not find file " + javaFileName);
         }
-
-        // Run codesign on the java binary.
         ProcessBuilder pb = new ProcessBuilder("codesign", "--display", "--verbose", javaFileName);
         pb.redirectErrorStream(true); // redirect stderr to stdout
         Process codesignProcess = pb.start();
+        return codesignProcess;
+    }
+
+    public static boolean hasPlistEntriesOSX() throws IOException {
```

Comment:
Although I understand why you chose this API name (it mimics the form used for `isHardenedOSX`), I don't think it is a good choice. `isHardenedOSX` is short for "is this a hardened OSX system". That mapping does not work with `hasPlistEntriesOSX`, which is more like "does this OSX system have plist entries". I would suggest `hasOSXPlistEntries`.